spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -78,13 +73,10 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		ConsumerProperties consumerProperties =
 				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
 		if (binder instanceof ExtendedPropertiesBinder) {
-			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
-			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
-			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
-			BeanUtils.copyProperties(consumerProperties, extendedConsumerProperties);
-			consumerProperties = extendedConsumerProperties;
+			consumerProperties = ProducerConsumerPropertiesUtil.getExtendedConsumerProperties((ExtendedPropertiesBinder) binder,
```

Comment:
I thought the same when I first started with this. But, the way binder is extracted is different in both the cases. Either we need to refactor that part or we can copy the same code twice (that extended properties extraction part) or have the code in common place like this PR has. I am ok with any of these options.